src/main/java/com/xceptance/xlt/report/util/JFreeChartUtils.java

```diff
@@ -935,6 +1004,88 @@ public static TimeSeries createMovingAverageTimeSeries(final TimeSeries series,
+     * 
+     * @param series
+     *            the source series
+     * @param percentage
+     *            the percentaged amount of values for building the moving average
+     * @return the time series
+     */
+    public static TimeSeries createMovingAverageTimeSeriesAdditonalPercentage(final TimeSeries series, final int percentage)
+    {
+        // take the last X percent of the values
+        final int samples = Math.max(2, series.getItemCount() * percentage / 100);
+
+        // derive the name from the source series
+        final String avgSeriesName = series.getKey() + " (Moving Average ("+ percentage +"%))";
+
+        return MovingAverage.createPointMovingAverage(series, avgSeriesName, samples);
+    }
+    
+    /**
+     *  Creates an additional, depending on given time, "moving average" time series from the given time series.
+     * 
+     * @param series
+     *            the source series
+     * @param value
+     *            the time value for building the moving average
+     * @param name
+     *            short name for time amount
+     * @return the time series
+     */ 
+    public static TimeSeries createMovingAverageTimeSeriesAdditonalTime(final TimeSeries series, final long value, String name)
+    {
+        // derive the name from the source series
+        final String avgSeriesName = series.getKey() + " (Moving Average Time ("+ name +"))";
+        int samples = 0;
+        
+        // check if we have items
+        if (series.getItemCount() > 0)
+        {        
+            TimeSeriesDataItem firstItem = series.getDataItem(0);
+            
+            long firstMillisecond = firstItem.getPeriod().getFirstMillisecond();
+            long time = firstMillisecond + value;
+            
+            for (int index = 0; index < series.getItemCount(); index++)
+            {
+                if (series.getDataItem(index).getPeriod().getFirstMillisecond() > time)
+                {
+                    samples = index;
+                    break;
+                }
+            }
+        }
+        return MovingAverage.createPointMovingAverage(series, avgSeriesName, Math.max(2,samples));
+    }
+    
+    /**
+     * Creates an additional, depending on amount of requests, "moving average" time series from the given time series.
+     * 
+     * @param series
+     *            the source series
+     * @param value
+     *            the amount of requests for building the moving average
+     * @param requestCount
+     *            overall amount of requests
+     * @return the time series
+     */
+    public static TimeSeries createMovingAverageTimeSeriesAdditonalRequests(final TimeSeries series, final int value, final double requestCount)
+    {
+        // percentage value from overall requests
+        int percentage = (int) Math.floor((value / requestCount) * 100);
+        // samples for chart data items
+        final int samples = Math.max(2, series.getItemCount() * percentage / 100);
```

Comment:
This doesn't seem to be correct as it uses a percentage of values instead of a fixed amount.  Maybe, I just don't get your intention here. But I'd expect something like the following
```
        final int samples = Math.max(2, value);
```

We might also have to verify that `value` is not greater than `requestCount`.